### PR TITLE
Port mathMLTree to @flow.

### PR DESCRIPTION
**Pull request by marcianx**

### CODE REVIEW

**Comment by edemaine** [comment]
@marcianx Did you mean to request me a reviewer? I'm not actually very familiar with Flow...

**Comment by marcianx** [comment]
Ah yes. I'm happy to reassign if you're not comfortable with it.

File: src/mathMLTree.js [hunk 1]
```diff
@@ -43,8 +61,8 @@ class MathNode {
             }
         }
 
-        for (let i = 0; i < this.children.length; i++) {
-            node.appendChild(this.children[i].toNode());
+        for (const child of this.children) {
```
Comment (kevinbarabash) [inline src/mathMLTree.js]: Happy to see `for-of`.

File: src/mathMLTree.js [hunk 2]
```diff
@@ -11,12 +12,29 @@
...
+ * https://developer.mozilla.org/en-US/docs/Web/MathML/Element.
+ */
+export type MathNodeType =
+    "math" | "annotation" | "semantics" |
+    "mtext" | "mn" | "mo" | "mi" | "mspace" |
+    "mover" | "munder" | "munderover" | "msup" | "msub" |
+    "mfrac" | "mroot" | "msqrt" |
+    "mtable" | "mtr" | "mtd" |
+    "mrow" | "menclose" |
+    "mstyle" | "mpadded" | "mphantom";
```
Comment (kevinbarabash) [inline src/mathMLTree.js]: Thankfully this list isn't too long.

**Comment by kevinbarabash** [comment]
Looks like there are some flow errors after updating the branch:
```
Error: src/functions/mod.js:81
 81:         const space = new mathMLTree.MathNode("mspace");
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ constructor call
 81:         const space = new mathMLTree.MathNode("mspace");
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ undefined (too few arguments, expected default/rest parameters). This type is incompatible with
 37:     constructor(type: MathNodeType, children: (MathNode | TextNode)[]) {
                                                    ^^^^^^^^^^^^^^^^^^^^^^ array type. See: src/mathMLTree.js:37
```

**Comment by marcianx** [comment]
Thanks. `children` should have been optional (already, the implementation assumed that and defaults to an empty array).

**Pull request merged**